go/kbfs/libkbfs/favorites.go

```diff
@@ -316,16 +319,25 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
 
 	kbpki := f.config.KBPKI()
 	// Fetch a new list if:
-	//  * The user asked us to refresh
-	//  * We haven't fetched it before
-	//  * It's stale
-	if (req.refresh || f.favCache == nil || f.config.Clock().Now().After(
-		f.cacheExpireTime)) && !req.clear {
-
+	//  (1) The user asked us to refresh
+	//  (2) We haven't fetched it before
+	//  (3) It's stale
+	//
+	// If just (3), use a short timeout so we can return the correct result
+	// quickly when offline.
+	needFetch := (req.refresh || f.favCache == nil) && !req.clear
+	wantFetch := f.config.Clock().Now().After(f.cacheExpireTime) && !req.clear
+	if needFetch || wantFetch {
+		getCtx := req.ctx
+		if !needFetch {
+			timeoutCtx, cancel := context.WithTimeout(req.ctx, favoritesServerTimeoutWhenCacheExpired)
```

Comment:
IMO this would be clearer as:
```go
if !needFetch {
	var cancel context.CancelFunc
	getCtx, cancel = context.WithTimeout(getCtx, ...)
	defer cancel()
}